### PR TITLE
GF-30 - Create Modal component and use it for sign up/in and log out

### PR DESCRIPTION
**Pull request by iamdavidpetric**

### CODE REVIEW

File: src/show/components/Modal/index.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
+import PropTypes from 'prop-types';
+
+const Modal = ({ children, setVisible, title, visible }) => {
+  return visible === true ? (
```
Comment (iamalextodoran) [inline src/show/components/Modal/index.js]:
```suggestion
  return visible ? (
```

File: src/show/components/Modal/index.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+              <h3 className='text-lg leading-6 font-medium text-gray-900'>
+                {title}
+              </h3>
+              {children}
+            </div>
+          </div>
+        </div>
+      </div>
+    </div>
+  ) : null;
```
Comment (iamalextodoran) [inline src/show/components/Modal/index.js]: &&

File: src/show/components/Modal/index.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+          className='fixed inset-0 bg-gray-500 bg-opacity-75 transition-opacity'
+          onClick={() => setVisible(false)}
+        />
+        <span className='inline-block align-middle h-screen'>&#8203;</span>
+        <div className='inline-block bg-white text-black rounded-lg text-left overflow-hidden shadow-xl transform transition-all my-8 align-middle max-w-lg w-full'>
+          <div className='flex justify-center p-6 pb-4'>
+            <div className='flex flex-col mt-3 text-center justify-center w-full'>
+              <h3 className='text-lg leading-6 font-medium text-gray-900'>
+                {title}
+              </h3>
```
Comment (iamalextodoran) [inline src/show/components/Modal/index.js]: title && 
```suggestion
              {title && <h3 className='text-lg leading-6 font-medium text-gray-900'>
                {title}
              </h3>}
```

File: src/show/components/Modal/index.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+          </div>
+        </div>
+      </div>
+    </div>
+  ) : null;
+};
+
+Modal.defaultProps = {
+  children: undefined,
+  setVisible: false,
```
Comment (iamalextodoran) [inline src/show/components/Modal/index.js]: this is a func.

File: src/show/components/Modal/index.js [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+  title: '',
+  visible: false
+};
+
+Modal.defaultProps = {
```
Comment (iamalextodoran) [inline src/show/components/Modal/index.js]: these are prop types, not default props.

File: src/show/components/Modal/index.js [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+  children: PropTypes.node.isRequired,
```
Comment (iamalextodoran) [inline src/show/components/Modal/index.js]: try again.

File: src/show/components/Modal/index.js [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+  setVisible: PropTypes.bool,
```
Comment (iamalextodoran) [inline src/show/components/Modal/index.js]: func.

File: src/show/components/Navbar/index.js [hunk 8]
```diff
@@ -1,63 +1,175 @@
...
+  const [openSignUpModal, setOpenSignUpModal] = useState(false);
+
+  useEffect(() => {
+    const listener = e => {
+      if (e.key === 'Escape') {
+        setOpenSignUpModal(null);
+      }
+    };
+    window.addEventListener('keydown', listener);
+  });
```
Comment (iamalextodoran) [inline src/show/components/Navbar/index.js]: This should be in the modal component.

File: src/show/components/Navbar/index.js [hunk 9]
```diff
@@ -1,63 +1,175 @@
...
+
+  const closeSignupModal = () => {
+    setOpenSignUpModal(false);
+  };
```
Comment (iamalextodoran) [inline src/show/components/Navbar/index.js]: No need to create a new function. Just use `setOpenSignUpModal(false)` where you use `closeSignupModal`

File: src/show/components/Navbar/index.js [hunk 10]
```diff
@@ -1,63 +1,175 @@
...
+          </div>
         </div>
+      </div>
+
+      <Modal setVisible={setOpenSignUpModal} visible={openSignUpModal}>
+        <div className='min-h-full flex items-center justify-center py-12 px-6'>
+          <div className='max-w-md w-full space-y-8'>
+            <div className='flex flex-col justify-center items-center'>
+              <BsFillPersonLinesFill
+                size={'3rem'}
```
Comment (iamalextodoran) [inline src/show/components/Navbar/index.js]: .

File: src/show/components/Navbar/index.js [hunk 11]
```diff
@@ -1,63 +1,175 @@
...
+                className='text-primary-500'
+              />
+              <h2 className='mt-2 text-center text-3xl font-extrabold text-gray-900'>
+                Create a new account
+              </h2>
+            </div>
+            <form className='mt-8 space-y-6' action='#' method='POST'>
+              <input type='hidden' name='remember' value='true' />
```
Comment (iamalextodoran) [inline src/show/components/Navbar/index.js]: what's this?

File: src/show/components/Navbar/index.js [hunk 12]
```diff
@@ -1,63 +1,175 @@
...
+            <div className='flex flex-col justify-center items-center'>
+              <BsFillPersonLinesFill
+                size={'3rem'}
+                className='text-primary-500'
+              />
+              <h2 className='mt-2 text-center text-3xl font-extrabold text-gray-900'>
+                Create a new account
+              </h2>
+            </div>
+            <form className='mt-8 space-y-6' action='#' method='POST'>
```
Comment (iamalextodoran) [inline src/show/components/Navbar/index.js]:
```suggestion
            <form className='mt-8 space-y-6' >
```

File: src/show/components/Navbar/index.js [hunk 13]
```diff
@@ -1,63 +1,175 @@
...
+                  </label>
+                  <input
+                    id='username'
+                    name='username'
+                    type='username'
+                    autoComplete='username'
+                    required
+                    className='appearance-none rounded-none relative block w-full px-3 py-2 border border-gray-300 placeholder-gray-500 text-gray-900 rounded-t-md focus:outline-none focus:ring-primary-500 focus:border-primary-500 focus:z-10 text-sm'
+                    placeholder='Username'
+                  />
```
Comment (iamalextodoran) [inline src/show/components/Navbar/index.js]: What about your textfield component?

File: src/show/components/Navbar/index.js [hunk 14]
```diff
@@ -1,63 +1,175 @@
...
+                  </label>
+                  <input
+                    id='email-address'
+                    name='email'
+                    type='email'
+                    autoComplete='email'
+                    required
+                    className='appearance-none rounded-none relative block w-full px-3 py-2 border border-gray-300 placeholder-gray-500 text-gray-900 focus:outline-none focus:ring-primary-500 focus:border-primary-500 focus:z-10 text-sm'
+                    placeholder='Email address'
+                  />
```
Comment (iamalextodoran) [inline src/show/components/Navbar/index.js]: TextField component

File: src/show/components/Navbar/index.js [hunk 15]
```diff
@@ -1,63 +1,175 @@
...
+                  <label htmlFor='confirmpassword' className='sr-only'>
+                    Confirm Password
+                  </label>
+                  <input
+                    id='confirmpassword'
+                    name='confirmpassword'
+                    type='password'
+                    autoComplete='current-password'
+                    required
+                    className='appearance-none rounded-none relative block w-full px-3 py-2 border border-gray-300 placeholder-gray-500 text-gray-900 rounded-b-md focus:outline-none focus:ring-primary-500 focus:border-primary-500 text-sm'
```
Comment (iamalextodoran) [inline src/show/components/Navbar/index.js]: Text field component every where.

File: src/show/components/Navbar/index.js [hunk 16]
```diff
@@ -1,63 +1,175 @@
...
+                </div>
+              </div>
+
+              <div className='flex items-center justify-between'></div>
+              <div>
+                <button
+                  type='submit'
+                  className='group relative w-full flex justify-center py-2 px-4 border border-transparent text-sm font-medium rounded-md text-white bg-primary-500 hover:bg-primary-600'>
+                  Sign Up
+                </button>
```
Comment (iamalextodoran) [inline src/show/components/Navbar/index.js]: You have a button component, same bellow.

File: src/show/components/Navbar/index.js [hunk 17]
```diff
@@ -1,63 +1,175 @@
...
+                <button
+                  onClick={closeSignupModal}
```
Comment (iamalextodoran) [inline src/show/components/Navbar/index.js]:
```suggestion
                  onClick={() => setShowSignUpModal(false)}
```

File: src/show/components/Navbar/index.js [hunk 18]
```diff
@@ -1,63 +1,175 @@
...
 import { FiUnlock } from 'react-icons/fi';
 import { TbStairsUp } from 'react-icons/tb';
 import { GiTreeGrowth } from 'react-icons/gi';
 import { useNavigate } from 'react-router-dom';
 import { MdOutlineExplore } from 'react-icons/md';
 import { IoConstructOutline } from 'react-icons/io5';
+import { BsFillPersonLinesFill } from 'react-icons/bs';
 
-import { Button } from '../';
+import { Fragment, useState, useEffect } from 'react';
```
Comment (iamalextodoran) [inline src/show/components/Navbar/index.js]: This should be grouped with the others above.

**Review by iamalextodoran — CHANGES_REQUESTED** [review]

**Comment by iamalextodoran** [comment]
Also PR title is off.

File: src/show/components/Modal/index.js (hunk 1 above)
Comment (iamdavidpetric) [reply src/show/components/Modal/index.js]: thx

File: src/show/components/Modal/index.js (hunk 2 above)
Comment (iamdavidpetric) [reply src/show/components/Modal/index.js]: my name is lightning mcqueen

File: src/show/components/Modal/index.js (hunk 3 above)
Comment (iamdavidpetric) [reply src/show/components/Modal/index.js]: pot face greseli mai rapid de cat poti tu corecta

File: src/show/components/Modal/index.js (hunk 4 above)
Comment (iamdavidpetric) [reply src/show/components/Modal/index.js]: sefu la greseli e aici

File: src/show/components/Modal/index.js (hunk 5 above)
Comment (iamdavidpetric) [reply src/show/components/Modal/index.js]: :(

File: src/show/components/Modal/index.js (hunk 7 above)
Comment (iamdavidpetric) [reply src/show/components/Modal/index.js]: esti smardoi nu gluma

File: src/show/components/Modal/index.js (hunk 6 above)
Comment (iamdavidpetric) [reply src/show/components/Modal/index.js]: no more mistakes on me (:

File: src/show/components/Navbar/index.js (hunk 8 above)
Comment (iamdavidpetric) [reply src/show/components/Navbar/index.js]: no more chinese work

File: src/show/components/Navbar/index.js (hunk 9 above)
Comment (iamdavidpetric) [reply src/show/components/Navbar/index.js]: good

File: src/show/components/Navbar/index.js (hunk 10 above)
Comment (iamdavidpetric) [reply src/show/components/Navbar/index.js]: hmm

File: src/show/components/Navbar/index.js (hunk 12 above)
Comment (iamdavidpetric) [reply src/show/components/Navbar/index.js]: 🇱🇷

File: src/show/components/Navbar/index.js (hunk 16 above)
Comment (iamdavidpetric) [reply src/show/components/Navbar/index.js]: you are right

File: src/show/components/Navbar/index.js (hunk 17 above)
Comment (iamdavidpetric) [reply src/show/components/Navbar/index.js]: already did it

File: src/show/components/Navbar/index.js (hunk 18 above)
Comment (iamdavidpetric) [reply src/show/components/Navbar/index.js]: ok :(

File: src/show/components/Navbar/index.js (hunk 11 above)
Comment (iamdavidpetric) [reply src/show/components/Navbar/index.js]: ??/

File: src/show/components/Navbar/index.js [hunk 19]
```diff
@@ -1,63 +1,221 @@
...
+                Log Out
+              </h2>
+            </div>
+            <form className='mt-8 space-y-6'>
+              <div className='flex items-center justify-between w-full'>
+                <div className='w-full'>
+                  <Button
+                    type='submit'
+                    label='Log Out'
+                    className='group relative w-full flex justify-center py-2 px-4 border border-transparent text-sm font-medium rounded-md text-white hover:text-white bg-primary-500 hover:bg-primary-600'></Button>
```
Comment (iamalextodoran) [inline src/show/components/Navbar/index.js]: Self closing.

File: src/show/components/Navbar/index.js [hunk 20]
```diff
@@ -1,63 +1,221 @@
...
+                  <Button
+                    onClick={() => setOpenLogOutModal(false)}
+                    label='Close'
+                    className='group relative w-full flex justify-center py-2 px-4 border border-transparent text-sm font-medium rounded-md text-white hover:text-white bg-primary-200 hover:bg-primary-400 mt-2'></Button>
```
Comment (iamalextodoran) [inline src/show/components/Navbar/index.js]: Self closing.

File: src/show/components/Navbar/index.js (hunk 19 above)
Comment (iamalextodoran) [reply src/show/components/Navbar/index.js]: You have a new button variant. Congrats! 🎉

File: src/show/components/Navbar/index.js (hunk 20 above)
Comment (iamalextodoran) [reply src/show/components/Navbar/index.js]: You have a new button variant. Congrats! 🎉

File: src/show/components/Navbar/index.js [hunk 21]
```diff
@@ -1,63 +1,221 @@
...
+                </div>
+              </div>
+            </form>
+          </div>
         </div>
-      </div>
-    </div>
+      </Modal>
+      <Modal setVisible={setOpenLogOutModal} visible={openLogOutModal}>
```
Comment (iamalextodoran) [inline src/show/components/Navbar/index.js:189]: Empty line before this so there's better readability.

File: src/show/components/Navbar/index.js [hunk 22]
```diff
@@ -1,63 +1,221 @@
...
+                  className='appearance-none rounded-none relative block w-full px-3 py-2 border border-gray-300 placeholder-gray-500 text-gray-900 rounded-b-md focus:outline-none focus:ring-primary-500 focus:border-primary-500 text-sm'
+                />
+              </div>
+
+              <div className='flex items-center justify-between w-full'>
+                <div className='w-full'>
+                  <Button
+                    type='submit'
+                    label='Sign In'
+                    className='group relative w-full flex justify-center py-2 px-4 border border-transparent text-sm font-medium rounded-md text-white hover:text-white bg-primary-500 hover:bg-primary-600'></Button>
```
Comment (iamalextodoran) [inline src/show/components/Navbar/index.js]: Self closing and use the new button variant.

File: src/show/components/Navbar/index.js [hunk 23]
```diff
@@ -1,63 +1,221 @@
...
+        <div className='min-h-full flex items-center justify-center py-12 px-6'>
+          <div className='max-w-md w-full space-y-8'>
+            <div className='flex flex-col justify-center items-center'>
+              <BsPersonCircle size='3rem' className='text-primary-500' />
+              <h2 className='mt-2 text-center text-3xl font-extrabold text-gray-900'>
+                Log in
+              </h2>
+            </div>
+            <form className='mt-8 space-y-6'>
+              <input type='hidden' name='remember' value='true' />
```
Comment (iamalextodoran) [inline src/show/components/Navbar/index.js]: Hidden input, what's the purpose?

File: src/show/components/Navbar/index.js [hunk 24]
```diff
@@ -1,63 +1,221 @@
...
 
-        <div className='flex transition-all duration-300 items-center hover:px-2 hover:py-1'>
-          <Button
-            variant='secondary'
-            iconLeft={<FiUnlock className='mx-1' size='1.5rem' />}
-            label='sign in'
-          />
+              <div
+                onClick={() =>
+                  setOpenLogInModal(true) & setOpenSignUpModal(false)
```
Comment (iamalextodoran) [inline src/show/components/Navbar/index.js]: What is this?

File: src/show/components/Navbar/index.js [hunk 25]
```diff
@@ -1,63 +1,221 @@
...
+              <div className='flex items-center justify-between w-full'>
+                <div className='w-full'>
+                  <Button
+                    type='submit'
+                    label='Sign Up'
+                    className='group relative w-full flex justify-center py-2 px-4 border border-transparent text-sm font-medium rounded-md text-white hover:text-white bg-primary-500 hover:bg-primary-600'></Button>
+                  <Button
+                    onClick={() => setOpenSignUpModal(false)}
+                    label='Close'
+                    className='group relative w-full flex justify-center py-2 px-4 border border-transparent text-sm font-medium rounded-md text-white hover:text-white bg-primary-200 hover:bg-primary-400 mt-2'></Button>
```
Comment (iamalextodoran) [inline src/show/components/Navbar/index.js]: Self closing and use the new button variant

File: src/show/components/Navbar/index.js [hunk 26]
```diff
@@ -1,63 +1,221 @@
...
+                  className='appearance-none rounded-none relative block w-full px-3 py-2 border border-gray-300 placeholder-gray-500 text-gray-900 rounded-b-md focus:outline-none focus:ring-primary-500 focus:border-primary-500 text-sm'
+                />
+              </div>
+
+              <div className='flex items-center justify-between w-full'>
+                <div className='w-full'>
+                  <Button
+                    type='submit'
+                    label='Sign Up'
+                    className='group relative w-full flex justify-center py-2 px-4 border border-transparent text-sm font-medium rounded-md text-white hover:text-white bg-primary-500 hover:bg-primary-600'></Button>
```
Comment (iamalextodoran) [inline src/show/components/Navbar/index.js]: Self closing and use the new variant.

File: src/show/components/Navbar/index.js [hunk 27]
```diff
@@ -1,63 +1,221 @@
...
+                  required
+                  type='email'
+                  placeholder='Email'
+                  className='appearance-none rounded-none relative block w-full px-3 py-2 border border-gray-300 placeholder-gray-500 text-gray-900 rounded-t-md focus:outline-none focus:ring-primary-500 focus:border-primary-500 focus:z-10 text-sm'
+                />
+                <TextField
+                  required
+                  type='password'
+                  placeholder='Confirm Password'
+                  className='appearance-none rounded-none relative block w-full px-3 py-2 border border-gray-300 placeholder-gray-500 text-gray-900 rounded-b-md focus:outline-none focus:ring-primary-500 focus:border-primary-500 text-sm'
```
Comment (iamalextodoran) [inline src/show/components/Navbar/index.js]: Use this classNames to style your text field. You should have one single styiling for your text fields.

File: src/show/components/Navbar/index.js [hunk 28]
```diff
@@ -1,63 +1,221 @@
...
+                  required
+                  type='password'
+                  placeholder='Password'
+                  className='appearance-none rounded-none relative block w-full px-3 py-2 border border-gray-300 placeholder-gray-500 text-gray-900  focus:outline-none focus:ring-primary-500 focus:border-primary-500 focus:z-10 text-sm'
+                />
+                <TextField
+                  required
+                  type='password'
+                  placeholder='Confirm Password'
+                  className='appearance-none rounded-none relative block w-full px-3 py-2 border border-gray-300 placeholder-gray-500 text-gray-900 rounded-b-md focus:outline-none focus:ring-primary-500 focus:border-primary-500 text-sm'
```
Comment (iamalextodoran) [inline src/show/components/Navbar/index.js]: Use this as your text field styling.

File: src/show/components/Navbar/index.js [hunk 29]
```diff
@@ -1,63 +1,221 @@
...
+                  required
+                  type='email'
+                  placeholder='Email'
+                  className='appearance-none rounded-none relative block w-full px-3 py-2 border border-gray-300 placeholder-gray-500 text-gray-900 focus:outline-none focus:ring-primary-500 focus:border-primary-500 focus:z-10 text-sm'
+                />
+                <TextField
+                  required
+                  type='password'
+                  placeholder='Password'
+                  className='appearance-none rounded-none relative block w-full px-3 py-2 border border-gray-300 placeholder-gray-500 text-gray-900  focus:outline-none focus:ring-primary-500 focus:border-primary-500 focus:z-10 text-sm'
```
Comment (iamalextodoran) [inline src/show/components/Navbar/index.js]: Use this as your text field styling.

File: src/show/components/Navbar/index.js [hunk 30]
```diff
@@ -1,63 +1,221 @@
...
+                  required
+                  type='username'
+                  placeholder='Username'
+                  className='appearance-none rounded-none relative block w-full px-3 py-2 border border-gray-300 placeholder-gray-500 text-gray-900 rounded-t-md focus:outline-none focus:ring-primary-500 focus:border-primary-500 focus:z-10 text-sm'
+                />
+                <TextField
+                  required
+                  type='email'
+                  placeholder='Email'
+                  className='appearance-none rounded-none relative block w-full px-3 py-2 border border-gray-300 placeholder-gray-500 text-gray-900 focus:outline-none focus:ring-primary-500 focus:border-primary-500 focus:z-10 text-sm'
```
Comment (iamalextodoran) [inline src/show/components/Navbar/index.js]: Use this as your styling.

File: src/show/components/Navbar/index.js [hunk 31]
```diff
@@ -1,63 +1,221 @@
...
+              </h2>
+            </div>
+            <form className='mt-8 space-y-6'>
+              <input type='hidden' name='remember' value='true' />
+              <div className='rounded-md shadow-sm -space-y-px'>
+                <TextField
+                  required
+                  type='username'
+                  placeholder='Username'
+                  className='appearance-none rounded-none relative block w-full px-3 py-2 border border-gray-300 placeholder-gray-500 text-gray-900 rounded-t-md focus:outline-none focus:ring-primary-500 focus:border-primary-500 focus:z-10 text-sm'
```
Comment (iamalextodoran) [inline src/show/components/Navbar/index.js]: Use this as your styling.

File: src/show/components/Navbar/index.js (hunk 11 above)
Comment (iamalextodoran) [reply src/show/components/Navbar/index.js]: Why do you need this? Don't resolve if it's not resolved.

File: src/show/components/Navbar/index.js [hunk 32]
```diff
@@ -1,63 +1,221 @@
...
+          </div>
 
-      <div className='flex flex-row items-center justify-end mx-2 w-1/3'>
-        <div className='flex transition-all duration-300 items-center hover:px-2 hover:py-1'>
-          <Button
-            variant='secondary'
-            iconLeft={<IoConstructOutline className='mx-1' size='1.5rem' />}
-            label='how it works'
-          />
+          {isLogin ? (
```
Comment (iamalextodoran) [inline src/show/components/Navbar/index.js]: `isLogin`? `isLoggedIn` maybe...

File: src/show/components/Navbar/index.js [hunk 33]
```diff
@@ -1,63 +1,221 @@
...
 import { EXPLORE_PATH, HOME_PATH } from '../../../process/routes/paths';
 
+import { Fragment, useState } from 'react';
+import TextField from '../TextField';
+
 const Navbar = () => {
   const navigate = useNavigate();
+  const [openSignUpModal, setOpenSignUpModal] = useState(false);
+  const [openLogInModal, setOpenLogInModal] = useState(false);
+  const [openLogOutModal, setOpenLogOutModal] = useState(false);
```
Comment (iamalextodoran) [inline src/show/components/Navbar/index.js]: Order length.

File: src/show/components/Navbar/index.js [hunk 34]
```diff
@@ -1,63 +1,221 @@
...
-
-import { Button } from '../';
+import {
+  BsArrowBarLeft,
+  BsFillPersonLinesFill,
+  BsPersonCircle
+} from 'react-icons/bs';
 import { EXPLORE_PATH, HOME_PATH } from '../../../process/routes/paths';
 
+import { Fragment, useState } from 'react';
```
Comment (iamalextodoran) [inline src/show/components/Navbar/index.js]: React should be grouped with react-icons and then ordedred by length.

File: src/show/components/Modal/index.js (hunk 7 above)
Comment (iamalextodoran) [reply src/show/components/Modal/index.js]: Don't resolve if it's not resolved. This is a func, not a bool.

File: src/show/components/Modal/index.js (hunk 4 above)
Comment (iamalextodoran) [reply src/show/components/Modal/index.js]: Don't resolve if it's not resolved. This is a function, not a bool.

**Review by iamalextodoran — CHANGES_REQUESTED** [review]

File: src/show/components/Navbar/index.js (hunk 34 above)
Comment (iamdavidpetric) [reply src/show/components/Navbar/index.js]: okay :)

File: src/show/components/Navbar/index.js (hunk 33 above)
Comment (iamdavidpetric) [reply src/show/components/Navbar/index.js]: no bun

File: src/show/components/Navbar/index.js (hunk 32 above)
Comment (iamdavidpetric) [reply src/show/components/Navbar/index.js]: biniuca

File: src/show/components/Navbar/index.js (hunk 11 above)
Comment (iamdavidpetric) [reply src/show/components/Navbar/index.js]: i don't need it

File: src/show/components/Navbar/index.js (hunk 26 above)
Comment (iamdavidpetric) [reply src/show/components/Navbar/index.js]: ooops

File: src/show/components/Navbar/index.js (hunk 25 above)
Comment (iamdavidpetric) [reply src/show/components/Navbar/index.js]: ooooops

File: src/show/components/Navbar/index.js (hunk 24 above)
Comment (iamdavidpetric) [reply src/show/components/Navbar/index.js]: no bine bulan

File: src/show/components/Navbar/index.js (hunk 23 above)
Comment (iamdavidpetric) [reply src/show/components/Navbar/index.js]: i am trying to hide from you

File: src/show/components/Navbar/index.js (hunk 22 above)
Comment (iamdavidpetric) [reply src/show/components/Navbar/index.js]: okay

File: src/show/components/Navbar/index.js (hunk 21 above)
Comment (iamdavidpetric) [reply src/show/components/Navbar/index.js:189]: i need a new pair of glasses

File: src/show/components/Navbar/index.js (hunk 19 above)
Comment (iamdavidpetric) [reply src/show/components/Navbar/index.js]: alright

File: src/show/components/Navbar/index.js (hunk 20 above)
Comment (iamdavidpetric) [reply src/show/components/Navbar/index.js]: alright 😮‍💨

File: src/show/components/Navbar/index.js (hunk 28 above)
Comment (iamdavidpetric) [reply src/show/components/Navbar/index.js]: yes sir

File: src/show/components/Navbar/index.js (hunk 29 above)
Comment (iamdavidpetric) [reply src/show/components/Navbar/index.js]: yes sir

File: src/show/components/Navbar/index.js (hunk 30 above)
Comment (iamdavidpetric) [reply src/show/components/Navbar/index.js]: yes sir

File: src/show/components/Navbar/index.js (hunk 31 above)
Comment (iamdavidpetric) [reply src/show/components/Navbar/index.js]: yes sir

File: src/show/components/Modal/index.js (hunk 4 above)
Comment (iamdavidpetric) [reply src/show/components/Modal/index.js]: nu mai fac, promit

File: src/show/components/Modal/index.js (hunk 7 above)
Comment (iamdavidpetric) [reply src/show/components/Modal/index.js]: nu mai fac sefu, numa nu ma lovi

File: src/show/components/Navbar/index.js [hunk 35]
```diff
@@ -1,63 +1,221 @@
-import { BiSearch } from 'react-icons/bi';
+import { ImExit } from 'react-icons/im';
 import { FiUnlock } from 'react-icons/fi';
+import { Fragment, useState } from 'react';
 import { TbStairsUp } from 'react-icons/tb';
 import { GiTreeGrowth } from 'react-icons/gi';
+import { Button, Modal, TextField } from '../';
```
Comment (iamalextodoran) [inline src/show/components/Navbar/index.js]: Not a library.

File: src/show/components/Navbar/index.js [hunk 36]
```diff
@@ -1,63 +1,221 @@
...
 import { MdOutlineExplore } from 'react-icons/md';
 import { IoConstructOutline } from 'react-icons/io5';
-
-import { Button } from '../';
+import {
+  BsArrowBarLeft,
+  BsFillPersonLinesFill,
+  BsPersonCircle
+} from 'react-icons/bs';
 import { EXPLORE_PATH, HOME_PATH } from '../../../process/routes/paths';
```
Comment (iamalextodoran) [inline src/show/components/Navbar/index.js:16]: Empty line before this, and group with components.

File: src/show/components/Navbar/index.js [hunk 37]
```diff
@@ -1,63 +1,221 @@
...
+                <div className='w-full'>
+                  <Button
+                    type='submit'
+                    variant='modal-button'
+                    label='Log Out'
+                    className='bg-primary-500 hover:bg-primary-600'
+                  />
+                  <Button
+                    onClick={() => setOpenLogOutModal(false)}
+                    variant='modal-button'
```
Comment (iamalextodoran) [inline src/show/components/Navbar/index.js]: Find a better variant name.

File: src/show/components/TextField/index.js [hunk 38]
```diff
@@ -1,14 +1,43 @@
...
       />
     </div>
   );
 };
 
+TextField.defaultProps = {
+  className: '',
+  onChange: () => {},
+  placeholder: '',
+  required: true,
```
Comment (iamalextodoran) [inline src/show/components/TextField/index.js]: make this false by default.

File: src/show/components/Navbar/index.js [hunk 39]
```diff
@@ -1,63 +1,221 @@
...
+              <BsFillPersonLinesFill size='3rem' className='text-primary-500' />
+              <h2 className='mt-2 text-center text-3xl font-extrabold text-gray-900'>
+                Create a new account
+              </h2>
+            </div>
+            <form className='mt-8 space-y-6'>
+              <div className='rounded-md shadow-sm -space-y-px'>
+                <TextField
+                  required
+                  type='username'
```
Comment (iamalextodoran) [inline src/show/components/Navbar/index.js]: There's no `username` type, should be text.

File: src/show/components/Button/index.js [hunk 40]
```diff
@@ -17,7 +17,9 @@ const Button = ({
       : 'border rounded-full border-primary-50 text-primary-50 hover:border-white hover:bg-primary-50 hover:text-white',
     'invert-outline': disabled
       ? 'border rounded-full border-green-300 text-greeen-300 border-white bg-green-300 py-1 text-white'
-      : 'border rounded-full border-green-500 text-greeen-500 border-white bg-green-500 py-1 text-white hover:bg-green-600'
+      : 'border rounded-full border-green-500 text-greeen-500 border-white bg-green-500 py-1 text-white hover:bg-green-600',
+    'modal-button':
```
Comment (iamalextodoran) [inline src/show/components/Button/index.js]: No disabled styling?

**Review by iamalextodoran — CHANGES_REQUESTED** [review]

File: src/show/components/Navbar/index.js (hunk 35 above)
Comment (iamdavidpetric) [reply src/show/components/Navbar/index.js]: hopefully this time is better

File: src/show/components/Navbar/index.js (hunk 36 above)
Comment (iamdavidpetric) [reply src/show/components/Navbar/index.js:16]: gud

File: src/show/components/TextField/index.js (hunk 38 above)
Comment (iamdavidpetric) [reply src/show/components/TextField/index.js]: true

File: src/show/components/Navbar/index.js (hunk 39 above)
Comment (iamdavidpetric) [reply src/show/components/Navbar/index.js]: text acceped

File: src/show/components/Navbar/index.js (hunk 37 above)
Comment (iamdavidpetric) [reply src/show/components/Navbar/index.js]: tertiary

File: src/show/components/Button/index.js (hunk 40 above)
Comment (iamdavidpetric) [reply src/show/components/Button/index.js]: :(

**Review by iamalextodoran — APPROVED** [review]

**Pull request merged**